tests/networkpolicy_test.go

```diff
@@ -0,0 +1,178 @@
+package tests
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	v1 "github.com/stackrox/rox/generated/api/v1"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/features"
+	"github.com/stackrox/rox/pkg/testutils"
+	"github.com/stretchr/testify/assert"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+const (
+	netpolDeploymentName = "nginx-deployment"
+)
+
+func getFakePolicyRequest(policyName, fieldName string) *v1.PostPolicyRequest {
+	return &v1.PostPolicyRequest{
+		Policy: &storage.Policy{
+			Id: "",
+			LifecycleStages: []storage.LifecycleStage{
+				storage.LifecycleStage_DEPLOY,
+			},
+			Name:               policyName,
+			IsDefault:          false,
+			CriteriaLocked:     false,
+			MitreVectorsLocked: false,
+			Severity:           storage.Severity_LOW_SEVERITY,
+			Categories:         []string{"Anomalous Activity"},
+			PolicySections: []*storage.PolicySection{
+				{
+					SectionName: "example",
+					PolicyGroups: []*storage.PolicyGroup{
+						{
+							FieldName:       fieldName,
+							BooleanOperator: 0,
+							Negate:          false,
+							Values: []*storage.PolicyValue{
+								{
+									Value: "true",
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		EnableStrictValidation: false,
+	}
+}
+
+func createPolicyIfMissing(t *testing.T, client v1.PolicyServiceClient, req *v1.PostPolicyRequest) (func(), error) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+	defer cancel()
+	p, err := client.PostPolicy(ctx, req)
+	if err != nil {
+		if s, ok := status.FromError(err); ok {
+			if s.Code() == codes.Internal && strings.Contains(s.Message(), "Could not add policy due to name validation") {
+				return func() {}, nil
+			}
+		}
+		return nil, err
+	}
+	return func() {
+		deletePolicy(t, client, p.Id)
+	}, nil
+}
+
+func deletePolicy(t *testing.T, client v1.PolicyServiceClient, id string) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+	defer cancel()
+	_, err := client.DeletePolicy(ctx, &v1.ResourceByID{Id: id})
+	assert.NoError(t, err)
+}
+
+func getAlertsForPolicy(t testutils.T, client v1.AlertServiceClient, policyName string) []*storage.ListAlert {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+	defer cancel()
+	var alerts []*storage.ListAlert
+	resp, err := client.ListAlerts(ctx, &v1.ListAlertsRequest{
+		Query: fmt.Sprintf("Policy:%s+Namespace:default", policyName),
+	})
+
+	if err != nil {
+		t.Errorf("Failed to get alerts: %s", err)
+		t.FailNow()
+	}
+
+	alerts = resp.GetAlerts()
+	return alerts
+}
+
+func CheckIfCentralHasFeatureFlag(t *testing.T, conn *grpc.ClientConn) bool {
+	ffClient := v1.NewFeatureFlagServiceClient(conn)
+
+	flags, err := ffClient.GetFeatureFlags(context.Background(), &v1.Empty{})
```

Comment:
I was thinking about the potential flakes for this, and thought that maybe we could use some client options, like for example: https://pkg.go.dev/google.golang.org/grpc#WaitForReady

Only if that is the source of flakes and only if it makes sense.

Reply:
Let's observe how this test reacts to running on CI. I think it should be fine how it is, but let's keep this in mind in case it become flaky.